contrib/flyway/src/FlywayModule.scala

```diff
@@ -0,0 +1,72 @@
+package mill
+package contrib.flyway
+
+import java.net.URLClassLoader
+
+import mill.scalalib.Lib.resolveDependencies
+import mill.scalalib._
+import org.flywaydb.core.Flyway
+import org.flywaydb.core.api.MigrationVersion
+import org.flywaydb.core.api.logging.LogFactory
+import org.flywaydb.core.internal.configuration.{ConfigUtils => flyway}
+import org.flywaydb.core.internal.info.MigrationInfoDumper
+import org.flywaydb.core.internal.logging.console.ConsoleLog.Level
+import org.flywaydb.core.internal.logging.console.ConsoleLogCreator
+import upickle.default.writer
+
+import scala.collection.JavaConverters._
+
+
+trait FlywayModule extends JavaModule {
+
+  def flywayUrl: T[String]
+  def flywayUser: T[String] = T("")
+  def flywayPassword: T[String] = T("")
+  //def flywayClassLocationDeps: T[Agg[Dep]] = Agg.empty[Dep]
+  //def flywayClassLocations: T[Seq[String]] = T(Nil)
+  def flywayFileLocations: T[Seq[PathRef]] = T(resources().map(pr => PathRef(pr.path / "db" / "migration", pr.quick)))
+  def flywayDriverDeps: T[Agg[Dep]]
+  def jdbcClasspath = T ( resolveDependencies(
+    repositories,
+    Lib.depToDependencyJava(_),
+    flywayDriverDeps()
+  ))
+
+  private def strToOptPair[A](key: String, v: String) =
+    Option(v)
+      .filter {
+        case a: String => a.nonEmpty
+        case _ => true
+      }
+      .map(key -> _)
+
+  private implicit val flywayWriter = writer[String].comap[Flyway](_.toString)
+  def flywayInstance = T.command {
```

Comment:
This command looks like a good candidate for a worker. http://www.lihaoyi.com/mill/page/tasks.html#workers